features/yield-calculator/YieldCalculator.tsx

```diff
@@ -0,0 +1,133 @@
+    PLACEHOLDER_PRICE
+  );
+  const [yieldAmount, setYieldAmount] = useState<number | null>(null);
+  const expirationTimestamp = empState.expirationTimestamp;
+
+  const calculateDaysToExpiry = () => {
+    if (expirationTimestamp) {
+      const currentTimestamp = Math.round(Date.now() / MS_PER_S);
+      const secondsToExpiry = expirationTimestamp.toNumber() - currentTimestamp;
+
+      return Math.round(secondsToExpiry / S_PER_DAY);
+    } else {
+      return null;
+    }
+  };
+
+  // @dev: We set this state var `daysToExpiry` after declaring `calculateDaysToExpiry()` because we first want to check if its value is
+  // non null in order to set its default value.
+  const [daysToExpiry, setDaysToExpiry] = useState<string | null>(
+    calculateDaysToExpiry()?.toString() || PLACEHOLDER_DAYS_TO_EXPIRY
+  );
+
+  const calculateYield = () => {
+    if (!tokenPrice || Number(tokenPrice) <= 0) {
+      return null;
+    }
+    if (!daysToExpiry || Number(daysToExpiry) <= 0) {
+      return null;
+    }
+
+    // `yieldPerUnit` = (FACE/yUSD_PX)^(1/(365/DAYS_TO_EXP)) - 1,
+    // where FACE = $1. More details: https://www.bankrate.com/glossary/a/apy-annual-percentage-yield/
+    const yieldPerUnit =
+      Math.pow(
+        1 / Number(tokenPrice),
+        1 / (Number(daysToExpiry) / DAYS_PER_YEAR)
+      ) - 1;
+    return yieldPerUnit;
+  };
+
+  const prettyPercentage = (x: number | null) => {
+    if (x === null) return "N/A";
+    return (x * 100).toFixed(4);
+  };
+
+  // Update the yield whenever the parameters change.
+  useEffect(() => {
+    setYieldAmount(calculateYield());
+  }, [tokenPrice, daysToExpiry]);
+
+  return (
+    <Box pt={4}>
+      <Container>
+        <form noValidate autoComplete="off">
+          <Typography variant="h5">yUSD Yield Calculator</Typography>
+          <FormInput></FormInput>
+          <FormInput>
+            <TextField
+              type="number"
+              label="Current yUSD Price (USD)"
+              value={tokenPrice?.toString() || ""}
+              onChange={(e) => setTokenPrice(e.target.value)}
+              variant="outlined"
+              InputLabelProps={{
+                shrink: true,
+              }}
+            />
+          </FormInput>
+          <FormInput>
+            <TextField
+              type="string"
+              label="Days to Expiry"
+              value={daysToExpiry?.toString() || ""}
+              onChange={(e) => setDaysToExpiry(e.target.value)}
+              helperText={`Days to expiry for selected EMP: ${calculateDaysToExpiry()}`}
+              variant="outlined"
+              InputLabelProps={{
+                shrink: true,
+              }}
+            />
+          </FormInput>
+          <FormInput>
+            <TextField
+              disabled
+              type="string"
+              label="APY (%)"
+              value={
+                yieldAmount !== null ? `${prettyPercentage(yieldAmount)}` : ""
```

Comment:
nit: this is the only place you use pretty percentage (I think). It returns "N/A" if the input is `null`. Why not just change it to return `""` if the input is null and avoid handling the null case in both places?

Reply:
Great catch. Done